# $DragonFly: src/sys/dev/netif/ath/hal/Attic/ah_osdep.c.patch,v 1.1 2006/07/13 09:15:22 sephe Exp $
--- ah_osdep.c	2006-04-22 17:06:43.000000000 +0800
+++ ah_osdep.c	2006-04-22 22:31:56.000000000 +0800
@@ -110,7 +110,7 @@ ath_hal_free(void* p)
 }
 
 void
-ath_hal_vprintf(struct ath_hal *ah, const char* fmt, va_list ap)
+ath_hal_vprintf(struct ath_hal *ah, const char* fmt, __va_list ap)
 {
 	vprintf(fmt, ap);
 }
@@ -118,16 +118,19 @@ ath_hal_vprintf(struct ath_hal *ah, cons
 void
 ath_hal_printf(struct ath_hal *ah, const char* fmt, ...)
 {
-	va_list ap;
-	va_start(ap, fmt);
+	__va_list ap;
+	__va_start(ap, fmt);
 	ath_hal_vprintf(ah, fmt, ap);
-	va_end(ap);
+	__va_end(ap);
 }
 
 const char*
 ath_hal_ether_sprintf(const u_int8_t *mac)
 {
-	return ether_sprintf(mac);
+	static char etherbuf[18];
+
+	snprintf(etherbuf, sizeof(etherbuf), "%6D", mac, ":");
+	return etherbuf;
 }
 
 #ifdef AH_DEBUG
@@ -136,9 +139,9 @@ HALDEBUG(struct ath_hal *ah, const char*
 {
 	if (ath_hal_debug) {
 		__va_list ap;
-		va_start(ap, fmt);
+		__va_start(ap, fmt);
 		ath_hal_vprintf(ah, fmt, ap);
-		va_end(ap);
+		__va_end(ap);
 	}
 }
 
@@ -364,10 +367,10 @@ ath_hal_delay(int n)
 u_int32_t
 ath_hal_getuptime(struct ath_hal *ah)
 {
-	struct bintime bt;
-	getbinuptime(&bt);
-	return (bt.sec * 1000) +
-		(((uint64_t)1000 * (uint32_t)(bt.frac >> 32)) >> 32);
+	struct timeval tv;
+
+	getmicrouptime(&tv);
+	return (tv.tv_sec * 1000) + (tv.tv_usec / 1000);
 }
 
 void
